common/types/src/main/java/io/helidon/common/types/DefaultAnnotationAndValue.java

```diff
@@ -177,31 +126,80 @@ public static DefaultAnnotationAndValue create(TypeName annoTypeName, Map<String
      * @return the result of the find
      */
     public static Optional<? extends AnnotationAndValue> findFirst(String annoTypeName,
-                                                         Collection<? extends AnnotationAndValue> coll) {
+                                                                   Collection<? extends AnnotationAndValue> coll) {
         assert (!annoTypeName.isBlank());
         return coll.stream()
                 .filter(it -> it.typeName().name().equals(annoTypeName))
                 .findFirst();
     }
 
-    @Override
-    public int compareTo(AnnotationAndValue other) {
-        return typeName().compareTo(other.typeName());
-    }
-
-
     /**
-     * Creates a builder for {@link io.helidon.pico.types.AnnotationAndValue}.
+     * Creates a builder for {@link AnnotationAndValue}.
      *
      * @return a fluent builder
      */
     public static Builder builder() {
         return new Builder();
     }
 
+    @Override
+    public String toString() {
+        String result = getClass().getSimpleName() + "(typeName=" + typeName();
+        Optional<String> value = value();
+        if (!values.isEmpty() && value.isEmpty()) {
+            result += ", values=" + values();
+        } else if (value.isPresent()) {
+            result += ", value=" + value.orElse(null);
+        }
+        return result + ")";
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(typeName(), values);
+    }
+
+    @Override
+    public boolean equals(Object another) {
+        if (!(another instanceof AnnotationAndValue)) {
+            return false;
+        }
+        if (!Objects.equals(typeName(), ((AnnotationAndValue) another).typeName())) {
+            return false;
+        }
+        if (!Objects.equals(values, ((DefaultAnnotationAndValue) another).values())) {
+            return false;
+        }
```

Comment:
Also if this is OK and this class cast exception can never be reached, it can be simplified as follows:
`return Objects.equals(values, ((DefaultAnnotationAndValue) another).values());`

Reply:
next push.